webapp/channels/src/components/threading/virtualized_thread_viewer/virtualized_thread_viewer.tsx

```diff
@@ -350,6 +354,14 @@ class ThreadViewerVirtualized extends PureComponent<Props, State> {
         const isLastPost = itemId === this.props.lastPost.id;
         const isRootPost = itemId === this.props.selected.id;
 
+        const post = useSelector((state: GlobalState) => getPost(state, this.props.selected.id));
+        const getThreadOrSynthetic = useMemo(makeGetThreadOrSynthetic, []);
+
+        const totalReplies = useSelector((state: GlobalState) => {
+            const thread = getThreadOrSynthetic(state, post);
+            return thread.reply_count || 0;
+        });
```

Comment:
I just came across this, and we can call these hooks inside a class component. ESLint doesn't catch this because `renderRow` is an arrow function (which is a bug https://github.com/facebook/react/issues/27431), but we need to fix this. I filed https://mattermost.atlassian.net/browse/MM-57292 to do that and with a suggestion for where to put this to also not have to prop drill the `replyCount` value down to where its used

Reply:
It's kind of a function component, but it's not really because it has a `this` as a field because it's a method of a class. It doesn't cause the app to crash like it would in the example I posted in the React repo because DynamicSizeList passes it into `createElement` instead of rendering it normally. There's definitely valid use cases for using `createElement` that way, but it feels like a coincidence that it allows you to get away with using hooks in an invalid way which is likely to break in the future.

As for a `makeRenderRow`, there's already a `Row` component so we shouldn't really need that